test/types/list_collections.test-d.ts

```diff
@@ -8,38 +8,30 @@ const db = new MongoClient('').db();
 type EitherCollectionInfoResult = CollectionInfo | Pick<CollectionInfo, 'name' | 'type'>;
 
 // We default to the CollectionInfo result type
-expectType<ListCollectionsCursor<Pick<CollectionInfo, 'name' | 'type'> | CollectionInfo>>(
-  db.listCollections()
-);
+expectType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
 // By default it isn't narrowed to either type
 expectNotType<ListCollectionsCursor<Pick<CollectionInfo, 'name' | 'type'>>>(db.listCollections());
-expectNotType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
+expectType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
 
 // Testing each argument variation
 db.listCollections();
 db.listCollections({ a: 2 });
 db.listCollections({ a: 2 }, { batchSize: 2 });
 
 const collections = await db.listCollections().toArray();
-expectType<EitherCollectionInfoResult[]>(collections);
+expectNotType<EitherCollectionInfoResult[]>(collections);
 
 const nameOnly = await db.listCollections({}, { nameOnly: true }).toArray();
 expectType<Pick<CollectionInfo, 'name' | 'type'>[]>(nameOnly);
 
 const fullInfo = await db.listCollections({}, { nameOnly: false }).toArray();
 expectType<CollectionInfo[]>(fullInfo);
 
-const couldBeEither = await db.listCollections({}, { nameOnly: Math.random() > 0.5 }).toArray();
-expectType<EitherCollectionInfoResult[]>(couldBeEither);
+const cannotBeEither = await db.listCollections({}, { nameOnly: Math.random() > 0.5 }).toArray();
+expectNotType<EitherCollectionInfoResult[]>(cannotBeEither);
+expectType<CollectionInfo[]>(cannotBeEither);
 
 // Showing here that:
 // regardless of the option the generic parameter can be used to coerce the result if need be
 // note the nameOnly: false, yet strings are returned
-const overridden = await db
-  .listCollections<Pick<CollectionInfo, 'name' | 'type'>>({}, { nameOnly: false })
-  .toArray();
-expectType<Pick<CollectionInfo, 'name' | 'type'>[]>(overridden);
-const overriddenWithToArray = await db
-  .listCollections({}, { nameOnly: false })
-  .toArray<Pick<CollectionInfo, 'name' | 'type'>>();
-expectType<Pick<CollectionInfo, 'name' | 'type'>[]>(overriddenWithToArray);
+// NODE-3468: No longer permits overriding of return type via a generic
```

Comment:
Can we remove the 3 comment lines above since they make no sense without the corresponding tests?